app/daos/ProfileDao.scala

```diff
@@ -270,15 +270,20 @@ class MariadbProfileDao @Inject()(val crewDao: MariadbCrewDao) extends ProfileDa
       case Some(crew) => {
         crewDao.findDBCrewModel(crew.id).flatMap(_ match {
           case Some (crewDB) => {
-            val action = for {
-              p  <- profiles.filter(prof => prof.email === profile.email)
-              s  <- supporters.filter(supporter => supporter.profileId === p.id)
-              sc <- supporterCrews.filter( suppCrew => suppCrew.crewId === crewDB.id && suppCrew.supporterId === s.id)
-            } yield sc.active
-            val updateAction = action.update(activeFlag)
-            dbConfig.db.run(updateAction).map(_ match {
-              case 1 => true
-              case _ => false
+            val supporterQuery = (for {
+              p <- profiles.filter(_.email === profile.email)
+              s <- supporters.filter(_.profileId === p.id)
+            } yield s)
+
+            dbConfig.db.run(supporterQuery.result).flatMap(_.headOption match {
+              case Some(supporterDB) => {
+                val updateQ = for { sc <- supporterCrews if sc.supporterId === supporterDB.id && sc.crewId === crewDB.id } yield (sc.active)
```

Comment:
You've created two database requests from one. :) Can you explain why, please?

Before there was a join from `profiles` to `supporters` to `supporterCrews` in only one query. Now, you have a first query joining from `profiles` to `supporters` and a second one that queries `supporterCrews` with the supporters ID as a parameter.

Reply:
No, okay, I understand. You want to make an `UPDATE` operation. Everything is clear now.